specifications/locks_auxiliary_vars/README.md

```diff
@@ -0,0 +1,48 @@
+# Two-way refinement using auxiliary variables
+
+Refinement proves that all behaviors implementation are a valid behavior of
+an abstract specification. It is usually not done from the abstract 
+specification to the implementation.
+
+This example shows an example of this second, unusual, case using auxiliary 
+variables, as presented in **Prophecy made simple** by *Lamport and Merz*.
+
+First a refinement from Peterson's algorithm to an abstract lock is provided
```

Comment:
What is Peterson's algorithm? Write a short introduction to it, the purpose of the algorithm, and its important properties & invariants.